frontend/scripts/metrica.js

```diff
@@ -0,0 +1,309 @@
+
+            device.ios = () => {
+                return device.iphone() || device.ipod() || device.ipad();
+            };
+
+            device.iphone = () => {
+                return !device.windows() && find('iphone');
+            };
+
+            device.ipod = () => {
+                return find('ipod');
+            };
+
+            device.ipad = () => {
+                return find('ipad');
+            };
+
+            device.android = () => {
+                return !device.windows() && find('android');
+            };
+
+            device.androidPhone = () => {
+                return device.android() && find('mobile');
+            };
+
+            device.androidTablet = () => {
+                return device.android() && !find('mobile');
+            };
+
+            device.blackberry = () => {
+                return find('blackberry') || find('bb10') || find('rim');
+            };
+
+            device.blackberryPhone = () => {
+                return device.blackberry() && !find('tablet');
+            };
+
+            device.blackberryTablet = () => {
+                return device.blackberry() && find('tablet');
+            };
+
+            device.windows = () => {
+                return find('windows');
+            };
+
+            device.windowsPhone = () => {
+                return device.windows() && find('phone');
+            };
+
+            device.windowsTablet = () => {
+                return device.windows() && (find('touch') && !device.windowsPhone());
+            };
+
+            device.fxos = () => {
+                return (find('(mobile;') || find('(tablet;')) && find('; rv:');
+            };
+
+            device.fxosPhone = () => {
+                return device.fxos() && find('mobile');
+            };
+
+            device.fxosTablet = () => {
+                return device.fxos() && find('tablet');
+            };
+
+            device.meego = () => {
+                return find('meego');
+            };
+
+            device.mobile = () => {
+                return device.androidPhone() ||
+                    device.iphone() ||
+                    device.ipod() ||
+                    device.windowsPhone() ||
+                    device.blackberryPhone() ||
+                    device.fxosPhone() ||
+                    device.meego();
+            };
+
+            device.tablet = () => {
+                return device.ipad() ||
+                    device.androidTablet() ||
+                    device.blackberryTablet() ||
+                    device.windowsTablet() ||
+                    device.fxosTablet();
+            };
+
+            device.desktop = () => {
+                return !device.tablet() && !device.mobile();
+            };
+
+            find = (needle) => {
+                return userAgent.indexOf(needle) !== -1;
+            };
+
+            return device;
+        }
+    };
+    try {
+        window.onload = () => {
```

Comment:
move handler to function (e.g. onDomReady). Then check if document.readyState !== 'complete' listen onload event, unless call function immediately. Because your script can be loaded both before page has loaded or after